include/wil/registry_basic.h

```diff
@@ -0,0 +1,1540 @@
+{
+    const reg_view_details::reg_view regview{key};
+    return regview.set_value(subkey, value_name, data);
+}
+
+inline void set_value_qword(HKEY key, _In_opt_ PCWSTR value_name, uint64_t data)
+{
+    return set_value_qword(key, nullptr, value_name, data);
+}
+
+inline void set_value_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ PCWSTR data)
+{
+    const reg_view_details::reg_view regview{key};
+    return regview.set_value(subkey, value_name, data);
+}
+
+inline void set_value_string(HKEY key, _In_opt_ PCWSTR value_name, _In_ PCWSTR data)
+{
+    return set_value_string(key, nullptr, value_name, data);
+}
+
+inline void set_value_multisz(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
+{
+    const reg_view_details::reg_view regview{key};
+    return regview.set_value_multisz(subkey, value_name, data);
+}
+
+inline void set_value_multisz(HKEY key, _In_opt_ PCWSTR value_name, const ::std::vector<::std::wstring>& data)
+{
+    return set_value_multisz(key, nullptr, value_name, data);
+}
+
+inline void set_value_multisz(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const ::std::list<::std::wstring>& data)
+{
+    const reg_view_details::reg_view regview{key};
+    return regview.set_value_multisz(subkey, value_name, data);
+}
+
+inline void set_value_multisz(HKEY key, _In_opt_ PCWSTR value_name, const ::std::list<::std::wstring>& data)
+{
+    return set_value_multisz(key, nullptr, value_name, data);
+}
+#endif
+
+inline HRESULT set_value_dword_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD data) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.set_value(subkey, value_name, data);
+}
+
+inline HRESULT set_value_dword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, DWORD data) WI_NOEXCEPT
+{
+    return set_value_dword_nothrow(key, nullptr, value_name, data);
+}
+
+inline HRESULT set_value_qword_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, uint64_t data) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.set_value(subkey, value_name, data);
+}
+
+inline HRESULT set_value_qword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, uint64_t data) WI_NOEXCEPT
+{
+    return set_value_qword_nothrow(key, nullptr, value_name, data);
+}
+
+inline HRESULT set_value_string_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ PCWSTR data) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.set_value(subkey, value_name, data);
+}
+
+inline HRESULT set_value_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In_ PCWSTR data) WI_NOEXCEPT
+{
+    return set_value_string_nothrow(key, nullptr, value_name, data);
+}
+
+// TODO: write multisz setters that don't throw.

```

Comment:
I had to disable these because they caused errors when compiling with clang. I believe they call functions which use `try/catch`, which isn't supported in exceptions-off scenarios.

Reply:
crap. yeah. I bet there are various sites which I didn't ifdef correctly for Exceptions or for STL support